vars/signArtifacts.groovy

```diff
@@ -97,33 +97,43 @@ void call(Map args = [:]) {
 
                 }
         }
-
     }
     else {
-        echo "PGP Signature Signing"
+        echo "PGP or Windows Signature Signing"
 
-        if( !fileExists("$WORKSPACE/sign.sh")) {
+        if (!fileExists("$WORKSPACE/sign.sh")) {
             git url: 'https://github.com/opensearch-project/opensearch-build.git', branch: 'main'
         }
 
         importPGPKey()
-        
+
         String arguments = generateArguments(args)
 
         // Sign artifacts
         withCredentials([usernamePassword(credentialsId: "${GITHUB_BOT_TOKEN_NAME}", usernameVariable: 'GITHUB_USER', passwordVariable: 'GITHUB_TOKEN')]) {
-            sh """
-                #!/bin/bash
-                set +x
-                export ROLE=${SIGNER_CLIENT_ROLE}
-                export EXTERNAL_ID=${SIGNER_CLIENT_EXTERNAL_ID}
-                export UNSIGNED_BUCKET=${SIGNER_CLIENT_UNSIGNED_BUCKET}
-                export SIGNED_BUCKET=${SIGNER_CLIENT_SIGNED_BUCKET}
-
-                $WORKSPACE/sign.sh ${arguments}
-            """
+            def configSecret = args.platform == "windows" ? "signer-windows-config" : "signer-pgp-config"
+            withCredentials([string(credentialsId: configSecret, variable: 'configs')]) {
```

Comment:
You can combine both `withCredentials` on line 113 and 115 into something like 
https://github.com/opensearch-project/opensearch-build/blob/main/vars/copyContainer.groovy#L24-L27

Also since we are renaming secrets can you name both as `jenkin-*`. We would want jenkins only to have secrets with `jenkins-*` prefixes.

Reply:
Combined in the latest commit. 

I thought we were saying all secrets in the account are all jenkins related. We still need add prefix?

Reply:
Had offline discussion with @prudhvigodithi  few days back. The account migh contain multiple AWS secrets, and we do not want to give jenkins access to all of them which can be restricted using prefixes. I have made this change in my recent PR.
Thanks!